client/selectors/index.js

```diff
@@ -277,3 +277,12 @@ export const isCurrentPlanningLockedInThisSession = createSelector(
         currentPlanning.lock_session === session.sessionId ? true : false
     )
 )
+
+export const isEventDetailLockedInThisSession = createSelector(
+    [getShowEventDetails, getEvents, getSessionDetails],
+    (showEventDetails, events, session) => {
+        const event = events[showEventDetails]
+        return event && event.lock_user === session.identity._id &&
+        event.lock_session === session.sessionId ? true : false
```

Comment:
Hmmm, I'm not convinced about the mystery behind this - it's pretty straight forward using just logical 'and' operators and a ternary operator in combination - I will consider changing this for your benefit though - no worries.